### PR TITLE
Bring back update notification

### PR DESCRIPTION
**Pull request by donker**
The current framework update service is broken. The proposed solution retrieves the last 5 releases from Github and checks the version of the first non-prerelease/RC version against the installed version. The result is cached and is shown to host users as a clear blue button prompting to upgrade. The button opens a new window with the release on Github visible.

Screenshot 2021-03-09 at 23 07 41

### CODE REVIEW

File: Dnn.AdminExperience/Library/Dnn.PersonaBar.UI/Services/ServerSummaryController.cs [hunk 1]
```diff
@@ -83,72 +85,48 @@ private bool GetVisibleSetting(string settingName)
...
+            }
+
+            return CBO.GetCachedObject<string>(new CacheItemArgs("DnnUpdateUrl"), this.RetrieveUpdateUrl);
+        }
+
+        private string RetrieveUpdateUrl(CacheItemArgs args)
+        {
+            try
+            {
+                var latestReleases = Globals.GetJsonObject<List<DTO.GithubLatestReleaseDTO>>("https://api.github.com/repos/dnnsoftware/dnn.platform/releases?per_page=5");
```
Comment (valadas) [inline Dnn.AdminExperience/Library/Dnn.PersonaBar.UI/Services/ServerSummaryController.cs:104]: Sould the url come from a config file maybe ?

File: Dnn.AdminExperience/Library/Dnn.PersonaBar.UI/admin/personaBar/scripts/main.js [hunk 2]
```diff
@@ -770,7 +770,7 @@ require(['jquery', 'knockout', 'moment', '../util', '../sf', '../config', './../
                         };
 
                         viewModel.updateText = ko.computed(function() {
-                            return viewModel.updateType() === 2 ? util.resx.PersonaBar.CriticalUpdate : util.resx.PersonaBar.NormalUpdate;
```
Comment (valadas) [inline Dnn.AdminExperience/Library/Dnn.PersonaBar.UI/admin/personaBar/scripts/main.js:773]: @mitchelsellers we used to have "Update" and "Critical Update", is that scenario still a supported use case ?

**Review by valadas — COMMENTED** [review]
I left 2 small comments, everything looks fine to me, the comments are mainly for discussion, other than discussion that I am fine with this change...

File: Dnn.AdminExperience/Library/Dnn.PersonaBar.UI/Services/ServerSummaryController.cs (hunk 1 above)
Comment (donker) [reply Dnn.AdminExperience/Library/Dnn.PersonaBar.UI/Services/ServerSummaryController.cs:104]: Although I am in favor of it, the truth is that people will never configure this. I.e. if we move to a different location, a new release and upgrade would hold the new values, but the older ones never will. So IMHO it is trying to solve something which we both see as a potential issue, but doesn't really solve it.

File: Dnn.AdminExperience/Library/Dnn.PersonaBar.UI/admin/personaBar/scripts/main.js (hunk 2 above)
Comment (donker) [reply Dnn.AdminExperience/Library/Dnn.PersonaBar.UI/admin/personaBar/scripts/main.js:773]: The code still exists, but in our current procedure I can't see (1) how we'd do it in Github and (2) if it is really useful/accurate/effective. Most of our upgrades are critical. But still few people upgrade ...

Comment (bdukes) [reply Dnn.AdminExperience/Library/Dnn.PersonaBar.UI/admin/personaBar/scripts/main.js:773]: Wouldn't normal vs. critical be based on security bulletins? We already have code in the Security extension of the Persona Bar to read that data. https://github.com/dnnsoftware/Dnn.Platform/blob/3e0d9c7c12e08f62ab458304aecdbc42b540d893/Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Services/SecurityController.cs#L608-L685

File: Dnn.AdminExperience/Library/Dnn.PersonaBar.UI/Services/ServerSummaryController.cs (hunk 1 above)
Comment (bdukes) [reply Dnn.AdminExperience/Library/Dnn.PersonaBar.UI/Services/ServerSummaryController.cs:104]: Are there concerns about rate limiting if we're accessing the API without a token?

**Review by mitchelsellers — CHANGES_REQUESTED** [review]

File: Dnn.AdminExperience/Library/Dnn.PersonaBar.UI/admin/personaBar/scripts/main.js (hunk 2 above)
Comment (valadas) [reply Dnn.AdminExperience/Library/Dnn.PersonaBar.UI/admin/personaBar/scripts/main.js:773]: yeah, that's what I was talking about just did not know where it came from in code...

**Comment by thabaum** [comment]
For the API rate limits, would a solution for this issue be a service created on a hosted site that caches hourly this information that is needed to connect to the link information provided by GitHub for the update?

**Comment by mitchelsellers** [comment]
@thabaum We have a solution ;)

**Review by mitchelsellers — APPROVED** [review]

**Comment by valadas** [comment]
@donker this looks good to me but I think we need to sync up with the changes in the service itself right? Let's discuss this in next meeting if you are available...

**Comment by donker** [comment]
> @donker this looks good to me but I think we need to sync up with the changes in the service itself right? Let's discuss this in next meeting if you are available...

Yes, but I'm assuming that is the easy part given we control everything

**Comment by mitchelsellers** [comment]
The UPdate Service changes have been deployed, this is good to go

**Pull request closed without merging**